compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs

```diff
@@ -203,7 +203,11 @@ impl IntoArgs for (CrateNum, SimplifiedType) {
 }
 
 provide! { tcx, def_id, other, cdata,
-    explicit_item_bounds => { table_defaulted_array }
+    explicit_item_bounds => {
+        let lazy = cdata.root.tables.explicit_item_bounds.get(cdata, def_id.index);
+        let output = if lazy.is_default() { &mut [] } else { tcx.arena.alloc_from_iter(lazy.decode((cdata, tcx))) };
+        ty::EarlyBinder(&*output)
```

Comment:
Can you leave a FIXME here that this code should be..... simplified?

Reply:
Yeah that's fair! In the newest commit, I pulled it into a new method on `CrateMetadataRef`, similar to what is done for some of the other queries below this in `cstore_impl.rs`. Please let me know if that's better.

I didn't add a FIXME on the new method but can if you think it'd be helpful. There's a lot of duplication around here in general between the `provide!` and `provide_one!` macros and the methods on `CrateMetadataRef` that maybe could be improved